packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapGroundOverlayController.m

```diff
@@ -0,0 +1,398 @@
+    } @catch (NSException *exception) {
+      @throw [NSException exceptionWithName:@"InvalidByteDescriptor"
+                                     reason:@"Unable to interpret bytes as a valid image."
+                                   userInfo:nil];
+    }
+  } else if ([bitmap isKindOfClass:[FGMPlatformBitmapAssetMap class]]) {
+    FGMPlatformBitmapAssetMap *bitmapAssetMap = bitmap;
+
+    image = [UIImage imageNamed:[registrar lookupKeyForAsset:bitmapAssetMap.assetName]];
+
+    if (bitmapAssetMap.bitmapScaling == FGMPlatformMapBitmapScalingAuto) {
+      NSNumber *width = bitmapAssetMap.width;
+      NSNumber *height = bitmapAssetMap.height;
+      if (width || height) {
+        image = [FLTGoogleMapGroundOverlayController scaledImage:image withScale:screenScale];
+        image = [FLTGoogleMapGroundOverlayController scaledImage:image
+                                                withWidth:width
+                                                   height:height
+                                              screenScale:screenScale];
+      } else {
+        image = [FLTGoogleMapGroundOverlayController scaledImage:image
+                                                withScale:bitmapAssetMap.imagePixelRatio];
+      }
+    }
+  } else if ([bitmap isKindOfClass:[FGMPlatformBitmapBytesMap class]]) {
+    FGMPlatformBitmapBytesMap *bitmapBytesMap = bitmap;
+    FlutterStandardTypedData *bytes = bitmapBytesMap.byteData;
+
+    @try {
+      image = [UIImage imageWithData:bytes.data scale:screenScale];
+      if (bitmapBytesMap.bitmapScaling == FGMPlatformMapBitmapScalingAuto) {
+        NSNumber *width = bitmapBytesMap.width;
+        NSNumber *height = bitmapBytesMap.height;
+
+        if (width || height) {
+          // Before scaling the image, image must be in screenScale.
+          image = [FLTGoogleMapGroundOverlayController scaledImage:image withScale:screenScale];
+          image = [FLTGoogleMapGroundOverlayController scaledImage:image
+                                                  withWidth:width
+                                                     height:height
+                                                screenScale:screenScale];
+        } else {
+          image = [FLTGoogleMapGroundOverlayController scaledImage:image
+                                                  withScale:bitmapBytesMap.imagePixelRatio];
+        }
+      } else {
+        // No scaling, load image from bytes without scale parameter.
+        image = [UIImage imageWithData:bytes.data];
+      }
+    } @catch (NSException *exception) {
+      @throw [NSException exceptionWithName:@"InvalidByteDescriptor"
+                                     reason:@"Unable to interpret bytes as a valid image."
+                                   userInfo:nil];
+    }
+  }
+
+  return image;
+}
+
+/// This method is deprecated within the context of `BitmapDescriptor.fromBytes` handling in the
```

Comment:
these all are duplicates from marker controller
can these be moved to separate helper class as static methods?

Reply:
Yes. Forgot about this one.